examples/custom-hooks/custom-hooks.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2022 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Examples showcasing the custom hooks API
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { registerReplaceHook } = require("@jazzer.js/hooking");
+
+/**
+ * An example of a bug detector hook.
+ */
+registerReplaceHook(
+	"JpegImage.jpegImage.constructor.prototype.copyToImageData.copyToImageData",
+	"jpeg-js",
+	false,
+	(thisPtr, params, hookId, origFn) => {
+		if (params[0].data[0] === 0) {
+			// we are only interested in image frames in which data[0] equals zero
+			throw Error(
+				"jpeg-js: copyToImageData() is called; image.data[0] equals 0"
+			);
+		}
+	}
+);
+
+/**
+ * An example of a pass-through hook.
+ * Calls the original function and returns the result without modification.
+ */
+registerReplaceHook(
+	"JpegImage.jpegImage.constructor.prototype.parse.parse.readUint16",
+	"jpeg-js",
+	false,
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
+	(thisPtr, params, hookId, origFn) => {
+		return origFn.apply(null, params);
+	}
+);
+
+/**
+ * An example of a fuzzing-enabling hook.
+ * The original function is never called.
+ * The hook does nothing other than logging.
+ * This can be useful for bypassing the fuzzing blockers to achieve
+ * coverage of more interesting functions.
+ */
+registerReplaceHook(
+	"JpegImage.jpegImage.buildHuffmanTable",
+	"jpeg-js",
+	false,
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
Disabling the warning is not required here, as there are no args, but in the first hook.